specification/iotoperations/IoTOperations.Management/models/dataflows/dataflows.tsp

```diff
@@ -212,7 +212,7 @@ model DataflowDestinationOperationSettings {
   endpointRef: string;
 
   @doc("Destination location, can be a topic or table name. Supports dynamic values with $topic, $systemProperties, $userProperties, $payload, $context, and $subscription.")
-  dataDestination: string;
+  dataDestination: nonemptystring;
```

Comment:
it would be good to sync with john, varun, and the other mq/dataflows folks to make sure we're checking for all the possible nonemptystrings across the resources